### PR TITLE
[CI] Validate conference data files

### PR DESCRIPTION
**Pull request by cketti**
Use [pyKwalify](https://github.com/Grokzen/pykwalify) to validate conference data files against a schema.

### CODE REVIEW

File: conference_schema.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        type:      date
+        required:  True
+        format:    "YYYY-MM-DD"
+      "end":
+        type:      date
+        required:  True
+        format:    "YYYY-MM-DD"
+      "site":
+        type:      str
+        required:  False
```
Comment (JakeWharton) [inline conference_schema.yml:43]: The template does not handle the case where this isn't available. If we want to make it optional we should not emit an anchor tag.

File: conference_schema.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    required:  True
+  "website":
+    type:      str
+    required:  True
+    pattern:   "https?://"
+  "location":
+    type:      str
+    required:  True
+  "status":
+    type:      str
```
Comment (JakeWharton) [inline conference_schema.yml:16]: Tempted to make this an enum or whatever so that we limit the values that it can have. I guess we can do that later.

**Review by JakeWharton — APPROVED** [review]
Got an example of what failure looks like?

File: conference_schema.yml (hunk 1 above)
Comment (cketti) [reply conference_schema.yml:43]: We fall back to the `website` value if the CfP site is missing.

https://github.com/AndroidStudyGroup/conferences/blob/01ed2a5bed7ce13cc2fe75d2cb24343f05ea7b5b/index.html#L29

Comment (JakeWharton) [reply conference_schema.yml:43]: Perfect

**Comment by cketti** [comment]
Example of a failure: https://github.com/cketti/conferences/runs/744316857?check_suite_focus=true#step:7:19

**Pull request merged**